### PR TITLE
[Security] Update keys for publicly verifiable veil developers

### PR DESCRIPTION
**Pull request by CaveSpectre11**
### Problem
PGP keys listed in https://github.com/Veil-Project/veil/tree/master/contrib/gitian-keys have not been updated from the bitcoin source.

### Solution
Keys that have been used to sign commits into the Veil core source were collected and those that have been loaded onto public keyservers, and match the users profile on keybase.io have been updated into the keys.txt file.  The readme has been rewritten as well to provide a little more information.

### Testing Results
When feeding the keys into your gpg keyring per the command in the readme, you will see the 4 keys process properly:
```
~/git-repos/veil/contrib/gitian-keys$ while read fingerprint keyholder_name; do gpg --keyserver hkp://keys.gnupg.net --recv-keys ${fingerprint}; done < ./keys.txt
gpg: key 413FA19FC334675F: "Cave Spectre <crypto@pontificatingnobody.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
gpg: key 6F55943B3D217111: "CodeofAllTrades <codeofalltrades@outlook.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
gpg: key 3BDCDA2D87A881D9: "Fuzzbawls <fuzzbawls@fuzzbawls.pw>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
gpg: key 062A341AB04CF5D7: "Mimir <mimir@runeledger.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```
**(Note that the output may look different if you are first loading these keys)**

Once the keys are loaded; you can use them to verify commits:

```
$ git verify-commit 25d176134
gpg: Signature made Mon 04 Nov 2019 07:20:49 PM EST
gpg:                using RSA key 6B5751C1D5544CB7425DCE5A413FA19FC334675F
gpg: Good signature from "Cave Spectre <crypto@pontificatingnobody.com>" [ultimate]
11
```

As noted in the README, we should actively be signing and trusting the developers with publicly accessible keys.  For example; you will see with @codeofalltrades 's key, before it's trusted:
```
$ git verify-commit a8d357d9c
gpg: Signature made Mon 21 Oct 2019 09:21:30 PM EDT
gpg:                using RSA key 3CABCD9E50469D76F40E010F6F55943B3D217111
gpg: Good signature from "CodeofAllTrades <codeofalltrades@outlook.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 3CAB CD9E 5046 9D76 F40E  010F 6F55 943B 3D21 7111
```
Trusting and signing:
```
$ gpg --edit-key code
gpg (GnuPG) 2.2.4; Copyright (C) 2017 Free Software Foundation, Inc.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.


pub  rsa4096/6F55943B3D217111
     created: 2019-09-07  expires: 2024-09-05  usage: SC
     trust: unknown       validity: unknown
sub  rsa4096/F14F9BA22AE9DCA4
     created: 2019-09-07  expires: 2024-09-05  usage: E
[ unknown] (1). CodeofAllTrades <codeofalltrades@outlook.com>

gpg> trust
pub  rsa4096/6F55943B3D217111
     created: 2019-09-07  expires: 2024-09-05  usage: SC
     trust: unknown       validity: unknown
sub  rsa4096/F14F9BA22AE9DCA4
     created: 2019-09-07  expires: 2024-09-05  usage: E
[ unknown] (1). CodeofAllTrades <codeofalltrades@outlook.com>

Please decide how far you trust this user to correctly verify other users' keys
(by looking at passports, checking fingerprints from different sources, etc.)

  1 = I don't know or won't say
  2 = I do NOT trust
  3 = I trust marginally
  4 = I trust fully
  5 = I trust ultimately
  m = back to the main menu

Your decision? 4

pub  rsa4096/6F55943B3D217111
     created: 2019-09-07  expires: 2024-09-05  usage: SC
     trust: full          validity: unknown
sub  rsa4096/F14F9BA22AE9DCA4
     created: 2019-09-07  expires: 2024-09-05  usage: E
[ unknown] (1). CodeofAllTrades <codeofalltrades@outlook.com>
Please note that the shown key validity is not necessarily correct
unless you restart the program.

gpg> sign

pub  rsa4096/6F55943B3D217111
     created: 2019-09-07  expires: 2024-09-05  usage: SC
     trust: full          validity: unknown
 Primary key fingerprint: 3CAB CD9E 5046 9D76 F40E  010F 6F55 943B 3D21 7111

     CodeofAllTrades <codeofalltrades@outlook.com>

This key is due to expire on 2024-09-05.
Are you sure that you want to sign this key with your
key "Cave Spectre <crypto@pontificatingnobody.com>" (413FA19FC334675F)

Really sign? (y/N) y
```
Afterwards you will see
```
$ git verify-commit a8d357d9c
gpg: Signature made Mon 21 Oct 2019 09:21:30 PM EDT
gpg:                using RSA key 3CABCD9E50469D76F40E010F6F55943B3D217111
gpg: checking the trustdb
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   6  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   6  signed:   1  trust: 0-, 0q, 0n, 1m, 5f, 0u
gpg: next trustdb check due at 2022-07-17
gpg: Good signature from "CodeofAllTrades <codeofalltrades@outlook.com>" [full]
```

### Additional Note
/contrib/verify-commits will need to be addressed as well; that can be done in a separate PR

### CODE REVIEW

**Comment by mimirmim** [comment]
I would keep the developers of Bitcoin on the list but comment separate Bitcoin from Veil.

File: contrib/gitian-keys/README.md [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 
 ```sh
-while read fingerprint keyholder_name; do gpg --keyserver hkp://subset.pool.sks-keyservers.net --recv-keys ${fingerprint}; done < ./keys.txt
+while read fingerprint keyholder_name; do gpg --keyserver hkp://keys.gnupg.net --recv-keys ${fingerprint}; done < ./keys.txt
 ```
 
-Add your key to the list if you provided Gitian signatures for two major or
-minor releases of Bitcoin Core.
+Developers for Veil should make sure they are signing their commits with their public
+key, and ensure the key is what is also loaded into your keybase.io profile.  When your
```
Comment (codeofalltrades) [inline contrib/gitian-keys/README.md]: Add https:// before keybase.io. Then it will be link.

Comment (CaveSpectre11) [reply contrib/gitian-keys/README.md]: Done.

**Comment by CaveSpectre11** [comment]
> I would keep the developers of Bitcoin on the list but comment separate Bitcoin from Veil.

Done.  Can't put comments in the file without tweaking the processing command in the readme, but I put a space in.  I changed it back to using the sks keyserver, since it processes the list that size much faster then the openpgp one I was using, and had all of them, where the other one was missing some.    I'll run the log again in the morning and see if there's others that can be added as well.

**Comment by codeofalltrades** [comment]
We will want to add sinetek.

**Comment by CaveSpectre11** [comment]
Since Sinetek isn't currently using it, and this isn't tied to any build (it's just live), it's quick to update.  Maybe push through for QA to give it a look over @codeofalltrades ?

**Pull request merged**